WorkloadManagementSystem/Utilities/ParametricJob.py

```diff
@@ -0,0 +1,164 @@
+""" Utilities to process parametric job definitions and generate
+    bunches of parametric jobs. It exposes the following functions:
+
+    getNumberOfParameters() - to get the total size of the bunch of parametric jobs
+    generateParametricJobs() - to get a list of expanded descriptions of all the jobs
+"""
+
+__RCSID__ = "$Id$"
+
+import re
+
+from DIRAC.Core.Utilities.ClassAd.ClassAdLight import ClassAd
+from DIRAC.Core.Utilities.ReturnValues import S_OK, S_ERROR
+from DIRAC.Core.Utilities.DErrno import EWMSJDL
+
+def __getParameterSequence( nPar, parList = [], parStart = 1, parStep = 0, parFactor = 1 ):
+
+  if parList:
+    if nPar != len( parList ):
+      return []
+    else:
+      parameterList = list( parList )
+  else:
+    # The first parameter must have the same type as the other ones even if not defined explicitly
+    parameterList = [ parStart * type( parFactor )( 1 ) + type( parStep )( 0 ) ]
+    for np in range( 1, nPar ):
+      parameterList.append( parameterList[np - 1] * parFactor + parStep )
+
+  return parameterList
+
+def getNumberOfParameters( jobClassAd ):
+  """ Get the number of parameters in the parametric job description
+
+  :param jobClassAd: ClassAd job description object
+  :return: int number of parameters, 0 if not a parametric job
+  """
+  if jobClassAd.lookupAttribute( 'Parameters' ):
+    if jobClassAd.isAttributeList( 'Parameters' ):
+      parameterList = jobClassAd.getListFromExpression( 'Parameters' )
+      return len( parameterList )
+    else:
+      return jobClassAd.getAttributeInt( 'Parameters' )
+  else:
+    return 0
+
+def __updateAttribute( classAd, attribute, parName, parValue ):
+
+  # If there is something to do:
+  pattern = '%%\(%s\)s' % parName
+  if parName == '0':
+    pattern = "%s"
+  expr = classAd.get_expression( attribute )
+  if not re.search( pattern, expr ):
+    return False
+
+  pattern = '%%(%s)s' % parName
+  if parName == '0':
+    pattern = "%s"
+
+  parValue = parValue.strip()
+  if classAd.isAttributeList( attribute):
+    parValue = parValue.strip()
+    if parValue.startswith( '{' ):
+      parValue = parValue[1:-1].strip()
```

Comment:
This is supposed to remove opening and closing curly brackets, right? Is it guaranteed that the last character is the closing bracket? why not use rstrip and lstrip for bracket and whitespace?